mesonbuild/compilers/cuda.py

```diff
@@ -17,6 +17,8 @@
 )
 from ..options import OptionKey
 from .compilers import Compiler
+import functools
+import operator
```

Comment:
These imports are out of place -- existing style in this file is to have stdlib imports at the top and application imports later.

Reply:
[`ruff rule I001`](https://docs.astral.sh/ruff/rules/unsorted-imports) would fix this but it was not applied because it currently finds 454 issues across the codebase.

Reply:
One can manually detail the fixups that were made, too. :)

In general isort is not used since it is incapable of *cleanly* doing it in an automated fashion (I opened a ticket about this in 2023). And ruff has made the decision that they won't even support grid-wrapped imports at all, so I'm totally uninterested in using ruff for this at all.

I'm a big believer that people should remember it's still possible to manually fix up their code when the automated rewrite tool isn't perfect.